package/gluon-hoodselector/check_site.lua

```diff
@@ -0,0 +1,37 @@
+need_string_match(in_domain({'domain_seed'}), '^' .. ('%x'):rep(64) .. '$')
+
+function need_nil(path)
+  need(path, function(val)
+    if val == nil then
+      return true
+    end
+    return false
+  end, true, "default hood should not contain shapes")
+  return nil
+end
+
+--Need to check if not default hood and does the default not contain shapes
+if this_domain() ~= need_string(in_site({'default_domain'})) then
+  local no_shapes = true
+  for _,shape in ipairs(need_table(in_domain({'hoodselector', 'shapes'}))) do
+    no_shapes = false
+    if #shape >= 2 then
+      for _, pos in ipairs(shape) do
+        if pos.lat == nil or not ( pos.lat < 90.0 and pos.lat > -90.0 ) then
+          need(in_domain({'hoodselector', 'shapes'}), function(err) return false end, true, "lat must match a range +/-90.0")
+        end
+        if pos.lon == nil or not ( pos.lon < 180.0 and pos.lon > -180.0 ) then
+          need(in_domain({'hoodselector', 'shapes'}), function(err) return false end, true, "lon must match a range +/-180.0")
+        end
+      end
+    end
+    if #shape < 2 then
+      need(in_domain({'hoodselector', 'shapes'}), function(err) return false end, true, "needs to have at least 2 coordinates for rectangular shapes.")
+    end
+  end
+  if no_shapes then
+    need(in_domain({'hoodselector', 'shapes'}), function(err) return false end, true, "no shapes are defined in hoods")
+  end
+else -- ente by default hood
+  need_nil(in_domain({'hoodselector', 'shapes'}))
+end
```

Comment:
This whole checker is much too complex:
* There is no reason to require the default domain not to have shapes (nothing breaks if it has)
* There is no reason to require all domains to have shapes (there can be domains that are only selected manually)
* You can drop the *in_domain* for most options, as `in_domain({'hoodselector', 'shapes'})` already ensures that hoodselector.shapes can only appear in domains
* How about adding a function *need_number_range* to the check_site_lib?
* You are using a weird workaround with a function that always returns false to generate custom error messages. Stop doing that.

Also, these configs are called "domains", not "hoods". The whole thing can be called "hoodselector", but error messages should only refer to "domains".

Reply:
> This whole checker is much too complex:
> 
>     * There is no reason to require the default domain not to have shapes (nothing breaks if it has)

The default domain represents the inverted surface of all shapes defined in the other domains.
So it simply doesn´t make sens that the default domain will have defined shapes.

>     * There is no reason to require all domains to have shapes (there can be domains that are only selected manually)

It will be not possible to select domains manually. The hoodselector aims to make the domains transparent.

>     * You can drop the _in_domain_ for most options, as `in_domain({'hoodselector', 'shapes'})` already ensures that hoodselector.shapes can only appear in domains

you mean all in_domain checks inside the for loop?

>     * How about adding a function _need_number_range_ to the check_site_lib?

Could be an option but maybe we can do that in an future MR.

> 
>     * You are using a weird workaround with a function that always returns false to generate custom error messages. Stop doing that.

Hm ? Where exactly I have done that? Maybe its a bug?

> Also, these configs are called "domains", not "hoods". The whole thing can be called "hoodselector", but error messages should only refer to "domains".

sure, I`ll change it.